engine/client.go

```diff
@@ -29,22 +30,23 @@ type (
 
 	// clientOptions holds all the configuration for the client
 	clientOptions struct {
-		cacheStore    *cacheStoreOptions    // Configuration options for Cachestore (ristretto, redis, etc.)
-		cluster       *clusterOptions       // Configuration options for the cluster coordinator
-		chainstate    *chainstateOptions    // Configuration options for Chainstate (broadcast, sync, etc.)
-		dataStore     *dataStoreOptions     // Configuration options for the DataStore (PostgreSQL, etc.)
-		debug         bool                  // If the client is in debug mode
-		encryptionKey string                // Encryption key for encrypting sensitive information (IE: paymail xPub) (hex encoded key)
-		httpClient    HTTPInterface         // HTTP interface to use
-		iuc           bool                  // (Input UTXO Check) True will check input utxos when saving transactions
-		logger        *zerolog.Logger       // Internal logging
-		metrics       *metrics.Metrics      // Metrics with a collector interface
-		models        *modelOptions         // Configuration options for the loaded models
-		newRelic      *newRelicOptions      // Configuration options for NewRelic
-		notifications *notificationsOptions // Configuration options for Notifications
-		paymail       *paymailOptions       // Paymail options & client
-		taskManager   *taskManagerOptions   // Configuration options for the TaskManager (TaskQ, etc.)
-		userAgent     string                // User agent for all outgoing requests
+		cacheStore       *cacheStoreOptions       // Configuration options for Cachestore (ristretto, redis, etc.)
+		cluster          *clusterOptions          // Configuration options for the cluster coordinator
+		chainstate       *chainstateOptions       // Configuration options for Chainstate (broadcast, sync, etc.)
+		dataStore        *dataStoreOptions        // Configuration options for the DataStore (PostgreSQL, etc.)
+		debug            bool                     // If the client is in debug mode
+		encryptionKey    string                   // Encryption key for encrypting sensitive information (IE: paymail xPub) (hex encoded key)
+		httpClient       HTTPInterface            // HTTP interface to use
+		iuc              bool                     // (Input UTXO Check) True will check input utxos when saving transactions
+		logger           *zerolog.Logger          // Internal logging
+		metrics          *metrics.Metrics         // Metrics with a collector interface
+		models           *modelOptions            // Configuration options for the loaded models
+		newRelic         *newRelicOptions         // Configuration options for NewRelic
+		notifications    *notificationsOptions    // Configuration options for Notifications
+		paymail          *paymailOptions          // Paymail options & client
+		transactionDraft *transactionDraftOptions // Draft transaction service
```

Comment:
Because of the `transactionDraftOptions` has only one field, why don't just put here the 
```golang
transactionDraftService draft.Service
```
?